### PR TITLE
[NET 6 Attribute Conversion] GameKit

### PR DESCRIPTION
**Pull request by tj-devel709**

### CODE REVIEW

File: src/GameKit/GameKit.cs [hunk 1]
```diff
@@ -21,19 +22,31 @@ namespace GameKit {
 #if !MONOMAC
 
 	// NSUInteger -> GKPeerPickerController.h
-#if XAMCORE_4_0
-	[NoTV] // preserve binary compatibility with existing/shipping code
```
Comment (tj-devel709) [inline src/GameKit/GameKit.cs:25]: Manually removing this XAMCORE_4_0 and [NoTV]
and throwing in the [UnsupportedOSPlatform ("tvos")] into the `#if NET`

File: src/GameKit/GameKit.cs [hunk 2]
```diff
@@ -147,31 +190,60 @@ public enum GKError : long {
...
 	[Native]
-#if WATCH && !XAMCORE_4_0
+#if WATCH
 	// removed in Xcode 10 but a breaking change (for us) to remove
 	[Obsolete ("Not used in watchOS.")]
 #else
 	[Unavailable (PlatformName.WatchOS)]
 	[ErrorDomain ("GKGameSessionErrorDomain")]
 #endif
+#endif // NET
```
Comment (tj-devel709) [inline src/GameKit/GameKit.cs:246]: Could I get extra eyes on this change above?
I was having trouble with GKGameSessionErrorCode not being exposed when I moved things around 
after the `#if WATCH`

Is any of this unnecessary?

Comment (rolfbjarne) [reply src/GameKit/GameKit.cs:246]: LGTM

**Review by rolfbjarne — APPROVED** [review]

File: src/GameKit/GameKit.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/GameKit/GameKit.cs:246]: same.

**Review by mandel-macaque — APPROVED** [review]

**Review by chamons — APPROVED** [review]

**Comment by tj-devel709** [comment]
**Unrelated Test Failures**
- monotouch-test/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2443 and

**Pull request merged**